src/ssl/bio.cc

```diff
@@ -131,7 +139,11 @@ Ssl::Bio::read(char *buf, int size, BIO *table)
 #if _SQUID_WINDOWS_
     const int result = socket_read_method(fd_, buf, size);
 #else
+    #if USE_OPENSSL_KTLS
```

Comment:
This topic might be outdated because of refactor.

Reply:
AFAICS, as of [124aa58](https://github.com/squid-cache/squid/pull/1629/commits/124aa58b201c3d25c0acccdc31178606f1cb820c) the refactor broke these things even worse than before.  Now when the feature is enabled-but-not-configured on Windows the entirely wrong `default_*_method` gets called instead of `socket_*_method`.

Reply:
Yes, the code should account for Windows possibly adding support. 
You could simplify the code a bit too with precompiler blocks, like so:
```
    int result;
#if OPENSSL_KTLS_SUPPORT
    if (BIO_next(table))
        result = BIO_write(BIO_next(table), buf, size);
    else
#endif
#if _SQUID_WINDOWS_
    result = socket_write_method(fd_, buf, size);
#else
    result = default_write_method(fd_, buf, size);
#endif
```